packages/eslint-plugin/typings/ts-eslint.d.ts

```diff
@@ -0,0 +1,416 @@
+    ): AST.Token[];
+
+    getTokenBefore(
+      node: TSESTree.Node | AST.Token | TSESTree.Comment,
+      options?: SourceCode.CursorWithSkipOptions
+    ): AST.Token | null;
+
+    getTokensBefore(
+      node: TSESTree.Node | AST.Token | TSESTree.Comment,
+      options?: SourceCode.CursorWithCountOptions
+    ): AST.Token[];
+
+    getTokenAfter(
+      node: TSESTree.Node | AST.Token | TSESTree.Comment,
+      options?: SourceCode.CursorWithSkipOptions
+    ): AST.Token | null;
+
+    getTokensAfter(
+      node: TSESTree.Node | AST.Token | TSESTree.Comment,
+      options?: SourceCode.CursorWithCountOptions
+    ): AST.Token[];
+
+    getFirstTokenBetween(
+      left: TSESTree.Node | AST.Token | TSESTree.Comment,
+      right: TSESTree.Node | AST.Token | TSESTree.Comment,
+      options?: SourceCode.CursorWithSkipOptions
+    ): AST.Token | null;
+
+    getFirstTokensBetween(
+      left: TSESTree.Node | AST.Token | TSESTree.Comment,
+      right: TSESTree.Node | AST.Token | TSESTree.Comment,
+      options?: SourceCode.CursorWithCountOptions
+    ): AST.Token[];
+
+    getLastTokenBetween(
+      left: TSESTree.Node | AST.Token | TSESTree.Comment,
+      right: TSESTree.Node | AST.Token | TSESTree.Comment,
+      options?: SourceCode.CursorWithSkipOptions
+    ): AST.Token | null;
+
+    getLastTokensBetween(
+      left: TSESTree.Node | AST.Token | TSESTree.Comment,
+      right: TSESTree.Node | AST.Token | TSESTree.Comment,
+      options?: SourceCode.CursorWithCountOptions
+    ): AST.Token[];
+
+    getTokensBetween(
+      left: TSESTree.Node | AST.Token | TSESTree.Comment,
+      right: TSESTree.Node | AST.Token | TSESTree.Comment,
+      padding?:
+        | number
+        | SourceCode.FilterPredicate
+        | SourceCode.CursorWithCountOptions
+    ): AST.Token[];
+
+    getTokens(
+      node: TSESTree.Node,
+      beforeCount?: number,
+      afterCount?: number
+    ): AST.Token[];
+    // eslint-disable-next-line no-dupe-class-members
+    getTokens(
+      node: TSESTree.Node,
+      options: SourceCode.FilterPredicate | SourceCode.CursorWithCountOptions
+    ): AST.Token[];
+
+    commentsExistBetween(
+      left: TSESTree.Node | AST.Token,
+      right: TSESTree.Node | AST.Token
+    ): boolean;
+
+    getCommentsBefore(
+      nodeOrToken: TSESTree.Node | AST.Token
+    ): TSESTree.Comment[];
+
+    getCommentsAfter(
+      nodeOrToken: TSESTree.Node | AST.Token
+    ): TSESTree.Comment[];
+
+    getCommentsInside(node: TSESTree.Node): TSESTree.Comment[];
+  }
+
+  //#endregion SourceCode
+
+  //#region Rule
+
+  interface RuleMetaData<TMessageIds extends string> {
+    /**
+     * True if the rule is deprecated, false otherwise
+     */
+    deprecated?: boolean;
+    /**
+     * Documentation for the rule
+     */
+    docs: {
+      /**
+       * The general category the rule falls within
+       */
+      category: 'Best Practices' | 'Stylistic Issues' | 'Variables';
```

Comment:
```suggestion
      category:
        | 'Best Practices'
        | 'Stylistic Issues'
        | 'Variables'
        | 'TSLint'
        | 'TypeScript';
```

Reply:
We can, it's a free text string! I was just enforcing values here so that our rules follow some standard.

I don't think it's worth having a `TSLint` or `Typescript` category.
The rules that we have put into those categories into the past would fit better into a more descriptive category name.

Reply:
i agree with this, but it should be done in follow up PR, after we decide about actual categories